ambry-store/src/test/java/com.github.ambry.store/StatsManagerTest.java

```diff
@@ -0,0 +1,300 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.Partition;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.PartitionState;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.StatsManagerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import com.github.ambry.utils.UtilsTest;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link StatsManager}.
+ */
+public class StatsManagerTest {
+  private static final long CAPACITY = 1024 * 1024 * 1024;
+  private static final int MAX_ACCOUNT_COUNT = 10;
+  private static final int MIN_ACCOUNT_COUNT = 5;
+  private static final int MAX_CONTAINER_COUNT = 6;
+  private static final int MIN_CONTAINER_COUNT = 3;
+  private final StatsManager statsManager;
+  private final String outputFileString;
+  private final File tempDir;
+  private final StatsSnapshot preAggregatedSnapshot;
+  private final Map<PartitionId, Store> storeMap;
+  private final Random random = new Random();
+  private final StatsManagerConfig config;
+
+  /**
+   * Deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws IOException {
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  public StatsManagerTest() throws IOException, StoreException {
+    tempDir = StoreTestUtils.createTempDirectory("nodeStatsDir-" + UtilsTest.getRandomString(10));
+    outputFileString = (new File(tempDir.getAbsolutePath(), "stats_output.json")).getAbsolutePath();
+    storeMap = new HashMap<>();
+    preAggregatedSnapshot = generateRandomSnapshot();
+    Pair<StatsSnapshot, StatsSnapshot> baseSliceAndNewSlice = new Pair<>(preAggregatedSnapshot, null);
+    for (int i = 0; i < 2; i++) {
+      baseSliceAndNewSlice = decomposeSnapshot(baseSliceAndNewSlice.getFirst());
+      storeMap.put(new Partition(i, PartitionState.READ_WRITE, CAPACITY),
```

Comment:
`Partition` is internal to the clustermap. Couldn't you simply mock `PartitionId` for the purpose of this test?